portfolio/src/main/webapp/script.js

```diff
@@ -20,28 +20,125 @@ import { photosData } from "./photos-data.js";
  * @return {html component}
  */
 function createPhoto(photo) {
-  const component = `<div class="col-4"> \
-          <figure class=\"figure\"> \
-              <img src=${photo.src} class=\"figure-img img-fluid rounded\" alt=${photo.location}> \
-              <div class=\"row figure-caption-container\"> \
-                  <figcaption class=\"figure-caption photo-location\">${photo.location}</figcaption> \
-                  <span class=\"dot\"></span> \
-                  <figcaption class=\"figure-caption photo-date\">${photo.date}</figcaption> \
-              </div> \
-          </figure> \
-      </div>`;
-
-  return component;
+  let component = [
+    "<div id=" + photo.epoch + ' class="col-4">',
+    '<figure class="figure">',
+    "<img src=" +
+      photo.src +
+      ' class="figure-img img-fluid rounded" alt=' +
+      photo.location +
+      ">",
+    '<div class="row figure-caption-container">',
+    '<figcaption class="figure-caption photo-location">' +
+      photo.location +
+      "</figcaption>",
+    '<span class="dot"></span>',
+    '<figcaption class="figure-caption photo-date">' +
+      photo.date +
+      "</figcaption>",
+    "</div>",
+    "</figure>",
+    "</div>",
+  ];
+  return $(component.join(""));
+}
+
+// dictionary mapping id : photo component
+const photoComponents = {};
+
+const allComponents = []
+
+const monthToNum = {
+  Jan: 0,
+  Feb: 1,
+  Mar: 2,
+  Apr: 3,
+  May: 4,
+  Jun: 5,
+  Jul: 6,
+  Aug: 7,
+  Sept: 8,
+  Oct: 9,
+  Nov: 10,
+  Dec: 11,
+};
+
+for (let index = 0; index < photosData.length; index++) {
+  let photo = photosData[index];
+
+  // add index id to each photo
+  photo.id = index;
+
+  // add epoch time to each photo
+  let date = photo.date.split(" ");
+  let year = date[2];
+  let month = monthToNum[date[0]];
+  let day = date[1].slice(0, -2);
```

Comment:
The reason why I didn't create a Date object is because my data inputs are like "Sept 4, 2020". I hard coded the photo data in photos-data.js, and I wanted the date to display on the website with the comma. This is probably a hacky way to do it. I imagine the right way to do it would be to store these photos in the back end?

Reply:
I was thinking that we could have the date display on the website with comma by converting the Date object to a string using:
```
const formatDate = (date) => {
  const month = numToMonth[date.getMonth()];
  return `${month} ${date.getDate()}, ${date.getFullYear()}`
}
```
and you already have epoch as `date.getTime()`

It's much safer to convert `Date` to a string than a string to a Date.
But, it's a lot of work at this point. I'm ok with your set up for now.